demos/link-prediction/knowledge-graphs/complex.ipynb

```diff
@@ -0,0 +1,708 @@
+      "text/plain": [
+       "<Figure size 504x576 with 2 Axes>"
+      ]
+     },
+     "metadata": {
+      "needs_background": "light"
+     },
+     "output_type": "display_data"
+    }
+   ],
+   "source": [
+    "utils.plot_history(wn18_history)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Evaluate the model\n",
+    "\n",
+    "We've now trained a model, so we can apply the evaluation procedure from the paper to it. This is done by taking each test edge `E = (s, r, o)`, and scoring it against all mutations `(s, r, n)` and `(n, r, o)` for every node `n` in the graph, that is, doing a prediction for every one of these edges similar to `E`. The \"raw\" rank is the number of mutated edges that have a higher predicted score than the true `E`."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 9,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "wn18_raw_ranks = ComplEx.rank_edges_against_all_nodes(\n",
+    "    wn18_model, wn18_gen.flow(wn18_test), wn18_g\n",
+    ")"
+   ]
+  },
+  {
+   "cell_type": "code",
```

Comment:
Minor optional thing, but the mis-alignment of the tables below is kinda jarring. Maybe you make and print out a dataframe with the papers results e.g.

```
pd.DataFrame({
    "mrr": [0.587], 
    "filtered mrr": [0.941], 
    "hits at 1": [0.936], 
    "hits at 3": [0.945],
    "hits at 10": [0.947],
})
```